seedVault/index.ts

```diff
@@ -0,0 +1,109 @@
+export type StorageAdapter = {
+  get(key: string): Promise<string> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+}
+
+export type CryptoUtilsAdapter =  {
+  encrypt(data: string, password: string): Promise<string>;
+  decrypt(data: string, password: string): Promise<string>;
+  hash(data: string, salt: string): Promise<string>;
+  generateRandomBytes(length: number): string;
+}
+
+export type SeedVaultConfig = {
+  secureStorageAdapter: StorageAdapter;
+  cryptoUtilsAdapter: CryptoUtilsAdapter;
+  commonStorageAdapter: StorageAdapter;
+};
+
+export enum SeedVaultStorageKeys {
+  PASSWORD_HASH = 'passwordHash',
+  PASSWORD_SALT = 'passwordSalt',
+  ENCRYPTED_KEY = 'encryptedKey',
+}
+
+class SeedVault {
+  private readonly _secureStorageAdapter: StorageAdapter;
+
+  private readonly _cryptoUtilsAdapter: CryptoUtilsAdapter;
+
+  private readonly _commonStorageAdapter: StorageAdapter;
+
+  VERSION = 1;
```

Comment:
I think we'll need to store this in common storage so we would know what version the stored data is in when doing an upgrade.